### PR TITLE
Search bot individuals

### PR DESCRIPTION
**Pull request by evamillan**
Adds the `isBot` filter to the individuals query, which accepts a boolean value and returns a list of individuals filtered by their bot status.

### CODE REVIEW

File: tests/test_schema.py [hunk 1]
```diff
@@ -1991,6 +2023,116 @@ def test_filter_registry_is_locked(self):
...
+                                email='jsmith@bitergia.com',
+                                username=None,
+                                source='mls',
+                                individual=indv)
+        Enrollment.objects.create(individual=indv, organization=org_ex)
+        Enrollment.objects.create(individual=indv, organization=org_bit,
+                                  start=datetime.datetime(1999, 1, 1, 0, 0, 0,
+                                                          tzinfo=dateutil.tz.tzutc()),
+                                  end=datetime.datetime(2000, 1, 1, 0, 0, 0,
+                                                        tzinfo=dateutil.tz.tzutc()))
```
Comment (sduenas) [inline tests/test_schema.py:2067]: We can remove some data that is not really needed on the test to make it easier to read.

File: tests/test_schema.py [hunk 2]
```diff
@@ -1991,6 +2023,116 @@ def test_filter_registry_is_locked(self):
...
+                                email='jdoe@example.com',
+                                username='jdoe',
+                                source='scm',
+                                individual=indv)
+        Identity.objects.create(uuid='B002',
+                                name=None,
+                                email='jdoe@libresoft.es',
+                                username=None,
+                                source='scm',
+                                individual=indv)
```
Comment (sduenas) [inline tests/test_schema.py:2086]: This can be removed too.

File: tests/test_schema.py [hunk 3]
```diff
@@ -1991,6 +2023,116 @@ def test_filter_registry_is_locked(self):
...
+
+        rol1 = enrollments[0]
+        self.assertEqual(rol1['organization']['name'], 'Bitergia')
+        self.assertEqual(rol1['start'], '1999-01-01T00:00:00+00:00')
+        self.assertEqual(rol1['end'], '2000-01-01T00:00:00+00:00')
+
+        rol2 = enrollments[1]
+        self.assertEqual(rol2['organization']['name'], 'Example')
+        self.assertEqual(rol2['start'], '1900-01-01T00:00:00+00:00')
+        self.assertEqual(rol2['end'], '2100-01-01T00:00:00+00:00')
```
Comment (sduenas) [inline tests/test_schema.py:2134]: This is not needed for the test.

File: tests/test_schema.py [hunk 4]
```diff
@@ -342,6 +342,38 @@
     }
   }
 }"""
+SH_INDIVIDUALS_BOT_FILTER = """{
```
Comment (sduenas) [inline tests/test_schema.py:345]: We should reduce the data that is requested in this query to make the test easier to read and understand it. We only need `mk` and `isBot` fields.

File: ui/src/components/Search.vue [hunk 5]
```diff
@@ -62,6 +62,8 @@ export default {
           const [filter, text] = value.split(":");
           if (filter === "lastUpdated") {
             this.parseLastUpdated(text);
+          } else if (text === "false") {
```
Comment (sduenas) [inline ui/src/components/Search.vue:65]: What happens if the `text` is `"true"`?

**Review by sduenas — CHANGES_REQUESTED** [review]
Please check my review.

**Comment by evamillan** [comment]
@sduenas Thanks for the feedback. I removed the unnecessary data and added a case for when the search text is "true".

File: tests/test_schema.py [hunk 6]
```diff
@@ -342,6 +342,16 @@
...
+  individuals(filters: {isBot: true}) {
```
Comment (sduenas) [inline tests/test_schema.py:346]: We should also test this for a `false` case.

**Review by sduenas — CHANGES_REQUESTED** [review]
I've been testing this patch using the UI and I don't get any error when I set invalid values for `isBot`. It always returns identities that are not bots. On the other hand, if I use the graphql API I get errors.

```
query individuals {
  individuals(filters: { lastUpdated:">=2020-11-25T18:04:38.000Z", isBot:"falsee" }) {
    entities {
      mk
      lastModified
    }
  }
}

{
  "errors": [
    {
      "message": "Argument \"filters\" has invalid value {lastUpdated: \">=2020-11-25T18:04:38.000Z\", isBot: \"falsee\"}.\nIn field \"isBot\": Expected type \"Boolean\", found \"falsee\".",
      "locations": [
        {
          "line": 2,
          "column": 24
        }
      ],
      "extensions": {
        "code": 128
      }
    }
  ]
}
```

Can you check it, please?

**Comment by evamillan** [comment]
> I've been testing this patch using the UI and I don't get any error when I set invalid values for `isBot`. It always returns identities that are not bots. On the other hand, if I use the graphql API I get errors.

@sduenas It looks like it converts the string to a boolean automatically when you query from the UI, but I can't find why it does that or how to prevent it. I've changed it so that it shows an error on the UI if the value is invalid and stops it from searching. Does this solution work for you?

**Comment by sduenas** [comment]
> > I've been testing this patch using the UI and I don't get any error when I set invalid values for `isBot`. It always returns identities that are not bots. On the other hand, if I use the graphql API I get errors.
> 
> @sduenas It looks like it converts the string to a boolean automatically when you query from the UI, but I can't find why it does that or how to prevent it. I've changed it so that it shows an error on the UI if the value is invalid and stops it from searching. Does this solution work for you?

I've been tracing the source of the problem. Python `graphene` doesn't parse correctly `string` values for `Boolean` fields. It uses `bool(value)` to cast to a boolean value but this is wrong because it always returns `True` for string values.
Your proposed solution is valid for us.

**Review by sduenas — APPROVED** [review]

**Pull request merged**